### PR TITLE
Automatic asset upload

### PR DESCRIPTION
**Pull request by dlactin**
## Because

- Uploading assets to the CDN is currently a manual process.

## This pull request

- Github action that will upload assets to stage/prod CDNs when changes to assets/product-icons assets/legal are merged to main. 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)

This has been tested in stage but the role has not been created in production yet.

### CODE REVIEW

File: .github/workflows/upload-assets-to-cdn.yml
Comment (jbuck) [inline .github/workflows/upload-assets-to-cdn.yml:26]: Hmm, I thought stage would be in us-east-1... or does this refer to sts?

**Review by jbuck — APPROVED** [review]

**Comment by jbuck** [comment]
@hwine tagging you for additional review, since this is the first time we've implemented this

File: .github/workflows/upload-assets-to-cdn.yml
Comment (dlactin) [reply .github/workflows/upload-assets-to-cdn.yml:26]: In this case the S3 bucket & IAM role but both are global, it complained without a region but changing it to us-east-1 for consistency might be good.

**Comment by hwine** [comment]
Like the other one, I'm going to need more context here.

File: .github/workflows/upload-assets-to-cdn.yml
Comment (hwine) [reply .github/workflows/upload-assets-to-cdn.yml:26]: S3 is "global-ish" -- if you want to receive S3 events, the receiver has to be in the same region the bucket was configured in. (I got bit on a lambda with this.) So using the origin region might preserve the hair of an sre/dev down the road 😁

Comment (dlactin) [reply .github/workflows/upload-assets-to-cdn.yml]: Sounds good to me! 

Updated stage region to us-east-1.

**Comment by dlactin** [comment]
I have created a slack app that is pending. Will need to create the slack secret when the app has been installed.

File: .github/workflows/upload-assets-to-cdn.yml
Comment (jbuck) [inline .github/workflows/upload-assets-to-cdn.yml:32]: To @hwine's notes, maybe this should be `sync` instead of `cp` (though we should very this won't do anything unexpected the first time we use this)

**Comment by jbuck** [comment]
@hwine also noted we should add CODEOWNERS for these files - we should check with the FxA folks who it'd make sense to put on that list for these files

**Comment by jbuck** [comment]
Current `main` branch protection rules: https://github.com/mozilla/fxa/settings/branch_protection_rules/4907005

We should consider enabling the `Require review from Code Owners ` rule

File: .github/workflows/upload-assets-to-cdn.yml
Comment (dlactin) [reply .github/workflows/upload-assets-to-cdn.yml:32]: I verified this works as intended on the stage bucket with a dryrun.

```
aws s3 sync --cache-control 'public,max-age=86400' --exclude "*" --include "*.pdf" --content-disposition attachment  assets/legal s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/legal --dryrun
(dryrun) upload: assets/legal/legal4.pdf to s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/legal/legal4.pdf
(dryrun) upload: assets/legal/test1.pdf to s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/legal/test1.pdf
(dryrun) upload: assets/legal/test2.pdf to s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/legal/test2.pdf
```

```
aws s3 sync --cache-control 'public,max-age=86400' --exclude "*" --include "*.svg" --include "*.png" assets/product-icons  s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/product-icons --dryrun
(dryrun) upload: assets/product-icons/test-icon1.svg to s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/product-icons/test-icon1.svg
(dryrun) upload: assets/product-icons/test-icon2.png to s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/product-icons/test-icon2.png
```

File: .github/workflows/upload-assets-to-cdn.yml
Comment (hwine) [inline .github/workflows/upload-assets-to-cdn.yml:45]: Has the business signed off on this? If they want old docs to still link, `cp` was the right thing. I was just asking a question :)

**Review by hwine — APPROVED** [review]
lgtm - ideally other branch protections would include:
- include administrators
- Dismiss stale pull request approvals when new commits are pushed

**Pull request merged**
